### ISSUE TITLE
Request for CMake 3.16 upgrade

### ISSUE THREAD

**Issue by forderud** [emscripten-core/emsdk#652]
The Emscripten docker image is currently using [CMake 3.13.4](https://packages.debian.org/buster/cmake), which is default for Debian Buster. I'm working on a codebase that require CMake 3.16 or newer, and would therefore appreciate if CMake could be upgraded to e.g. [3.16.3](https://packages.debian.org/buster-backports/cmake) as included in `buster-backports`.

Updating the Dockerfile with the following commands should do the trick:
```
# Activate backports for more recent version of CMake
echo "deb http://deb.debian.org/debian buster-backports main" > /etc/apt/sources.list.d/backports.list
apt-get update
apt-get -t buster-backports -y install cmake # from https://packages.debian.org/buster-backports/cmake
```

**Comment by forderud** [comment]
https://github.com/emscripten-core/emsdk/issues/652

**Comment by sbc100** [comment]
Would it not make sense to create your own docker file based on ours and extent it with those lines? 

On side note, requiring 3.16 is going going to make the upstream software hart to port in general.   Debian buster was released just a few months ago and cmake 3.16 was released about a year ago on Nov 26, 2019.   Have you considered suggesting that upstream support older build systems?  Is there some killer features of 3.16 that they really can't live without?  

None of this is to say that we can't do this.. but its worth pushing back a little to avoid overcomplicating things.

**Comment by forderud** [comment]
> Would it not make sense to create your own docker file based on ours and extent it with those lines?

You're of course right. Sorry for not thinking about that possibility.

> On side note, requiring 3.16 is going going to make the upstream software hart to port in general. Debian buster was released just a few months ago and cmake 3.16 was released about a year ago on Nov 26, 2019. Have you considered suggesting that upstream support older build systems? Is there some killer features of 3.16 that they really can't live without?

The codebase in question require CMake 3.16 due to usage of `target_precompile_headers` to speed up the build process. Precompiled headers makes a big difference on the build times, so I don't think it's realistic to change.

> None of this is to say that we can't do this.. but its worth pushing back a little to avoid overcomplicating things.

I fully understand your reluctance. Please feel free to close the request if you think it will complicate things at your end. I can instead apply the CMake upgrade on my end.

**Comment by kleisauke** [comment]
fwiw, PR https://github.com/emscripten-core/emsdk/pull/747 could help with this, if you need CMake >= 3.16.
